refaster-runner/src/main/java/tech/picnic/errorprone/refaster/runner/Refaster.java

```diff
@@ -118,12 +123,52 @@ private static void applyMatches(
       ImmutableRangeSet<Integer> ranges = getReplacementRanges(description, endPositions);
       if (ranges.asRanges().stream().noneMatch(replacedSections::intersects)) {
         /* This suggested fix does not overlap with any ("larger") replacement seen until now. Apply it. */
-        state.reportMatch(description);
+        state.reportMatch(augmentDescription(description, getSeverityOverride(state)));
         replacedSections.addAll(ranges);
       }
     }
   }
 
+  private Optional<SeverityLevel> getSeverityOverride(VisitorState state) {
+    return Optional.ofNullable(state.errorProneOptions().getSeverityMap().get(canonicalName()))
+        .flatMap(Refaster::toSeverityLevel);
+  }
+
+  private static Optional<SeverityLevel> toSeverityLevel(Severity severity) {
+    switch (severity) {
+      case DEFAULT:
+        return Optional.empty();
+      case WARN:
+        return Optional.of(WARNING);
+      case ERROR:
+        return Optional.of(ERROR);
+      default:
+        throw new IllegalStateException(String.format("Unsupported severity='%s'", severity));
+    }
+  }
+
+  /**
+   * Updates the given {@link Description}'s details such that {@link
+   * VisitorState#reportMatch(Description)} will override the reported severity only if this bug
+   * checker's severity was explicitly configured.
+   *
+   * <p>The original check name (i.e. the Refaster template name) is prepended to the {@link
+   * Description}'s message. The replacement check name ("Refaster Template") is chosen such that it
+   * is guaranteed not to match any canonical bug checker name (as that could cause {@link
+   * VisitorState#reportMatch(Description)}} to override the reported severity).
+   */
+  private static Description augmentDescription(
+      Description description, Optional<SeverityLevel> severityOverride) {
+    return Description.builder(
+            description.position,
+            "Refaster Rule",
```

Comment:
We use both `Refaster Rule` and `Refaster Template` a lot. Not really sure if we need to settle on using one of the two. 
Now it depends a bit on the situation, I feel.

Reply:
We use both `Refaster Rule` and `Refaster Template` a lot. Not really sure if we need to settle on using one of the two. 
Now it depends a bit on the situation, I feel.

To be honest, I try to only use the word "template" to avoid confusing and make people use the same words. I'd be okay with switching everything to template. WDYT?

Reply:
Can't find a clear distinction between a rule and a template in the [original documentation](https://errorprone.info/docs/refaster) either. I'm happy with both (slightly prefer template, given the annotations), as long as we are consistent :+1:

Reply:
Yeah, this has bugged me for a while as well. I _suspect_ it's more correct to call "the whole thing" a Refaster _rule_, since (a) it's represented by the `RefasterRule` type and (b) the individual methods are called before/after _templates_.

So that would make "Refaster Rule" correct for this code.

That still leaves many incorrect usages elsewhere in the code base... maybe something for a separate PR?

Reply:
Yea let's defer to another PR. 

I'm also in camp let's be consistent as @Badbond is haha. I bet you are as well. 
It makes slightly more sense to say "lets add a rule for that" instead of "lets add a template for that".